pkg/client/restclient/request.go

```diff
@@ -105,16 +106,13 @@ type Request struct {
 	resource     string
 	resourceName string
 	subresource  string
-	selector     labels.Selector
 	timeout      time.Duration
 
 	// output
 	err  error
 	body io.Reader
 
-	// The constructed request and the response
-	req  *http.Request
-	resp *http.Response
+	ctx context.Context
```

Comment:
> In the future, if we start passing context.Context throughout more of our code, it could facilitate per-request local tracing, per-request credentials, and other things.

Yeah, that's what worries me.  The "magic container" aspect of the context can make it difficult to track what's actually going to happen.  You could still construct a context to do this inside of `request.go`, but I don't think we need to expose it and have people try to figure out what to do with or start using it to stuff various values into.